### PR TITLE
feat(logs): print branch pristine status as part of the branch summary

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes
Title ☝

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by Gabriel-Ladzaretti** [comment]
```
DEBUG: Branch summary (repository=g-ladzaretti/renovate-testing)
       "cacheModified": true,
       "baseBranches": [{"branchName": "main", "sha": "f4860622a7bd6190c92f65309b587ea950970d9a"}],
       "branches": [
         {
           "branchName": "renovate/bunyan-1.x",
           "branchSha": "3af1e870e985b00d1ea90a0312e34c8778b192cd",
           "baseBranch": "main",
           "baseBranchSha": "f4860622a7bd6190c92f65309b587ea950970d9a",
           "automerge": false,
           "isModified": false,
           "isPristine": true
         },
         {
           "branchName": "renovate/got-12.x",
           "branchSha": "b74e6e345c218b8b88e1c0cf6963881f0dc37013",
           "baseBranch": "main",
           "baseBranchSha": "f4860622a7bd6190c92f65309b587ea950970d9a",
           "automerge": false,
           "isModified": false,
           "isPristine": true
         }
       ],

```

File: lib/workers/repository/finalise/repository-statistics.ts [hunk 1]
```diff
@@ -42,11 +42,12 @@ export function runRenovateRepoStats(
 
 function branchCacheToMetadata({
   branchName,
-  sha: branchSha,
   baseBranch,
   baseBranchSha,
   automerge,
   isModified,
+  sha: branchSha,
```
Comment (rarkins) [inline lib/workers/repository/finalise/repository-statistics.ts:49]: Why reorder if they're not alphabetically sorted anyway?

Comment (Gabriel-Ladzaretti) [reply lib/workers/repository/finalise/repository-statistics.ts:49]: only to group together literal renames, aesthetic wise looks better to me

```ts
function branchCacheToMetadata({
  branchName,
  baseBranch,
  baseBranchSha,
  automerge,
  isModified,
  sha: branchSha,
  pristine: isPristine,
}: BranchCache): BranchMetadata {}
```

vs

```ts
function branchCacheToMetadata({
  branchName,
  sha: branchSha,
  baseBranch,
  baseBranchSha,
  automerge,
  isModified,
  pristine: isPristine
}: BranchCache): BranchMetadata {}
```

Comment (Gabriel-Ladzaretti) [reply lib/workers/repository/finalise/repository-statistics.ts]: now sorted as well which ticks both boxes

**Review by rarkins — APPROVED** [review]

**Pull request merged**